plugin/documents.py

```diff
@@ -501,6 +501,7 @@ def _do_highlights_async(self) -> None:
 
     def _on_highlights(self, response: Optional[List]) -> None:
         if not response:
+            self._clear_highlight_regions()
```

Comment:
Could have one clear at the beginning of the function since the other branch also does that.